lib/kubernetes-deploy/kubernetes_resource/daemon_set.rb

```diff
@@ -24,7 +24,8 @@ def deploy_succeeded?
     end
 
     def deploy_failed?
-      pods.present? && pods.any?(&:deploy_failed?)
+      pods.present? && pods.any?(&:deploy_failed?) &&
+      observed_generation == current_generation
```

Comment:
should this not be `!=`? Or maybe I just misunderstood it

Reply:
As I understand it, and @KnVerey will hopefully correct me if I'm wrong, `current_generation ` gets bumped when we `apply` the config. `observed_generation ` gets set by the controller reporting the status. When they don't match it means that we're looking at a status for the previous config not current. E.g. we don't want to say the deployment has failed or succeeded before the status reflects the current config's state.

Reply:
oh! Thanks for the clarification